inference/models/owlv2/owlv2.py

```diff
@@ -56,15 +56,79 @@ def _check_size_limit(self):
                 self.popitem(last=False)
 
 
+def preprocess_image(
+    np_image: np.ndarray,
+    image_size: Tuple[int, int],
+    image_mean: torch.Tensor,
+    image_std: torch.Tensor,
+) -> torch.Tensor:
+    """Preprocess an image for OWLv2 by resizing, normalizing, and padding it.
+    This is much faster than using the Owlv2Processor directly, as we ensure we use GPU if available.
+
+    Args:
+        np_image (np.ndarray): The image to preprocess, with shape (H, W, 3)
+        image_size (tuple[int, int]): The target size of the image
+        image_mean (torch.Tensor): The mean of the image, on DEVICE, with shape (1, 3, 1, 1)
+        image_std (torch.Tensor): The standard deviation of the image, on DEVICE, with shape (1, 3, 1, 1)
+
+    Returns:
+        torch.Tensor: The preprocessed image, on DEVICE, with shape (1, 3, H, W)
+    """
+    current_size = np_image.shape[:2]
+
+    r = min(image_size[0] / current_size[0], image_size[1] / current_size[1])
+    target_size = (int(r * current_size[0]), int(r * current_size[1]))
+
+    torch_image = (
+        torch.tensor(np_image)
+        .permute(2, 0, 1)
+        .unsqueeze(0)
+        .to(DEVICE)
+        .to(dtype=torch.float32)
+        / 255.0
+    )
+    torch_image = F.interpolate(
+        torch_image, size=target_size, mode="bilinear", align_corners=False
+    )
+
+    padded_image_tensor = torch.ones((1, 3, *image_size), device=DEVICE) * 0.5
+    padded_image_tensor[:, :, : torch_image.shape[2], : torch_image.shape[3]] = (
+        torch_image
+    )
+
+    padded_image_tensor = (padded_image_tensor - image_mean) / image_std
+
+    return padded_image_tensor
+
+
 class OwlV2(RoboflowCoreModel):
     task_type = "object-detection"
     box_format = "xywh"
 
     def __init__(self, *args, model_id="owlv2/owlv2-base-patch16-ensemble", **kwargs):
         super().__init__(*args, model_id=model_id, **kwargs)
         hf_id = os.path.join("google", self.version_id)
-        self.processor = Owlv2Processor.from_pretrained(hf_id)
+        processor = Owlv2Processor.from_pretrained(hf_id)
```

Comment:
I can't see where the `processor` is initialized, it was `self.processor` before - is this change intentional?